### DISCUSSION TITLE
Tag name not well-formed

### DISCUSSION THREAD

**Discussion by Imane0** [community/community#26551] [Actions]
<p>In the root of my repository, I have a VERSION.txt file that contains something like v1.2.3.</p>
<p>My workflow is:</p>
<pre><code class="lang-auto">name: My workflow name

on:
  push:
    branches: [ master ]

env:
  MAJOR: $(cat VERSION.txt | cut -d . -f 1)
  MINOR: $(cat VERSION.txt | cut -d . -f 2)
  PATCH: $(cat VERSION.txt | cut -d . -f 3)

jobs:

  linting_unit_tests:
    runs-on: ubuntu-latest
    steps:
    - name: Checkout master branch
      uses: actions/checkout@v2

    - name: Install Python3
      uses: actions/setup-python@v2
      with:
        python-version: '3.9'
    - name: Install requirements
      run:  pip install -r requirements.txt

    - name: Linting with sort
      uses: jamescurtin/isort-action@master

    - name: Linting with black
      run: black --check .

    - name: Run tests
      run: |
          python -m pytest tests/

  push_docker_image:
    needs: [linting_unit_tests]
    runs-on: ubuntu-latest
    outputs:
      version: ${{ steps.get_version.outputs.version }}

    steps:
    - uses: actions/checkout@v2

    - name: Get docker image version
      id: get_version
      run:  |
        VERSION="v${MAJOR}.${MINOR}.${PATCH}"
        echo "::set-output name=version::${VERSION}"

    - name: Build docker image
      run:  |
        git tag ${{ steps.get_version.outputs.version }}
        cd docker
        docker-compose build

    - name: Tag docker image
      run:  |
        VERSION=${{ steps.get_version.outputs.version }}
        docker tag "my-app:latest" "us.gcr.io/${{secrets.GCP_PROJECT_ID}}/my-app:$VERSION"

    - name: Setup Cloud SDK
      uses: google-github-actions/setup-gcloud@master
      with:
        project_id: ${{ secrets.GCP_PROJECT_ID }}
        service_account_key: ${{ secrets.GCP_SA_KEY }}
        export_default_credentials: true

    - name: Authenticate docker with gcp
      run: |
        gcloud auth configure-docker

    - name: Push docker image
      run: |
        VERSION=${{ steps.get_version.outputs.version }}
        docker push "us.gcr.io/${{secrets.GCP_PROJECT_ID}}/my-app:$VERSION

  create_release:
    needs: [push_docker_image]
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2

    - name: Echo release name
      run: |
        echo ${{ needs.push_docker_image.outputs.version }} # this outputs the right name for the release

    - name: Create release
      uses: actions/create-release@v1
      env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      tag_name: ${{ needs.push_docker_image.outputs.version }} # THIS THROWS ERROR
      release_name: ${{ needs.push_docker_image.outputs.version }}
       draft: false
       prerelease: false
</code></pre>
<p>The step “Echo release name” outputs the right name I want to give to the release, which is something like v1.2.3<br>
However, the “Create release” step throws an error:</p>
<pre><code class="lang-auto">Error: Validation Failed: {"resource":"Release","code":"custom","field":"tag_name","message":"tag_name is not a valid tag"}, {"resource":"Release","code":"custom","field":"tag_name","message":"tag_name is not well-formed"}, {"resource":"Release","code":"custom","message":"Published releases must have a valid tag"}
</code></pre>
<p>It seems like <code>tag_name</code> is taking the value <code>v$(cat VERSION.txt | cut -d . -f 1).$(cat VERSION.txt | cut -d . -f 2).$(cat VERSION.txt | cut -d . -f 3)</code> instead of the value <code>v1.2.3</code></p>
<p>How to make <code>tag_name</code> take the value v1.2.3 ?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The problem is with this <code>env</code> definition, you can’t use Bash expressions there:</p>
<aside class="quote no-group" data-username="Imane0" data-post="1" data-topic="178815">
<div class="title">
<div class="quote-controls"></div>
 Imane0:</div>
<blockquote>
<pre><code class="lang-yaml">env:
  MAJOR: $(cat VERSION.txt | cut -d . -f 1)
  MINOR: $(cat VERSION.txt | cut -d . -f 2)
  PATCH: $(cat VERSION.txt | cut -d . -f 3)
</code></pre>
</blockquote>
</aside>
<p>You’ll need to process the version file in a step, and set <a href="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-output-parameter">output parameters</a> or <a href="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable">environment variables</a>, whichever works better for your use case. If you need the version parts in multiple jobs it’s probably better to use outputs and turn them into job outputs (see <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs"><code>jobs.&lt;job_id&gt;.outputs</code></a>).</p>

**Comment by Imane0** [comment]
<p>Thank you for your reply. I removed the <code>env</code> key at the beginning of the yaml file, and replaced the <code>get_version</code> step with the following:</p>
<pre><code class="lang-auto">   - name: Get docker image version
      id: get_version
      run:  |
        MAJOR=$(cat VERSION.txt | cut -d . -f 1)
        MINOR=$(cat VERSION.txt | cut -d . -f 2)
        PATCH=$(cat VERSION.txt | cut -d . -f 3)
        VERSION="v${MAJOR}.${MINOR}.${PATCH}"
        echo "::set-output name=version::${VERSION}"
</code></pre>
<p>and it worked.</p>
